packages/graphql/src/schema-model/parser/parse-annotation.ts

```diff
@@ -18,100 +18,23 @@
+        groupedDirectives.set(directive.name.value, [...directivesOfName, directive]);
+    });
 
-function parseDirective(directive: DirectiveNode): Annotation | undefined {
-    switch (directive.name.value) {
-        case AnnotationsKey.authentication:
-            return parseAuthenticationAnnotation(directive);
-        case AnnotationsKey.authorization:
-            return parseAuthorizationAnnotation(directive);
-        case AnnotationsKey.coalesce:
-            return parseCoalesceAnnotation(directive);
-        case AnnotationsKey.customResolver:
-            return parseCustomResolverAnnotation(directive);
-        case AnnotationsKey.cypher:
-            return parseCypherAnnotation(directive);
-        case AnnotationsKey.default:
-            return parseDefaultAnnotation(directive);
-        case AnnotationsKey.filterable:
-            return parseFilterableAnnotation(directive);
-        case AnnotationsKey.fulltext:
-            return parseFullTextAnnotation(directive);
-        case AnnotationsKey.id:
-            return new IDAnnotation();
-        case AnnotationsKey.jwtClaim:
-            return parseJWTClaimAnnotation(directive);
-        case AnnotationsKey.jwtPayload:
-            return parseJWTPayloadAnnotation(directive);
-        case AnnotationsKey.mutation:
-            return parseMutationAnnotation(directive);
-        case AnnotationsKey.plural:
-            return parsePluralAnnotation(directive);
-        case AnnotationsKey.populatedBy:
-            return parsePopulatedByAnnotation(directive);
-        case AnnotationsKey.private:
-            return parsePrivateAnnotation(directive);
-        case AnnotationsKey.query:
-            return parseQueryAnnotation(directive);
-        case AnnotationsKey.limit:
-            return parseLimitAnnotation(directive);
-        case AnnotationsKey.selectable:
-            return parseSelectableAnnotation(directive);
-        case AnnotationsKey.settable:
-            return parseSettableAnnotation(directive);
-        case AnnotationsKey.subscription:
-            return parseSubscriptionAnnotation(directive);
-        case AnnotationsKey.subscriptionsAuthorization:
-            return parseSubscriptionsAuthorizationAnnotation(directive);
-        case AnnotationsKey.timestamp:
-            return parseTimestampAnnotation(directive);
-        case AnnotationsKey.unique:
-            return parseUniqueAnnotation(directive);
-        case AnnotationsKey.relayId:
-            return new RelayIDAnnotation();
-        default:
-            return undefined;
-    }
+    const result: Partial<Annotations> = {};
+    Object.entries(annotationsParsers).forEach(([name, parser]) => {
+        const relevantDirectives = groupedDirectives.get(name);
+        if (relevantDirectives?.length) {
+            const [first] = relevantDirectives;
+            result[name] = parser(first!, relevantDirectives);
+        }
+    });
```

Comment:
As before change it to `for...of` (alternatively, maybe a `.reduce` could be a bit more succinct here)

There are also a couple of things I'd like to discuss about this logic. Overall I think this is a good change compared to the previous version, and some of them are minor points that may not require changes, but would like to know your opinion on these:

* This code assumes that the directive name (`directive.name.value`) is going to match the annotation name. I think that is true at the moment, and we can live with this, but it would be a bit better to make that either explicit through types or to decouple that need (if it is easy enough to do so, as I said, this is good enough)

A small type improvement that I could propose here:

```ts
const relevantDirectives = groupedDirectives.get(name) ?? [];
const firstDirective = relevantDirectives[0];
if (firstDirective) {
  result[name] = parser(first, relevantDirectives);
}
```

Functionally, it is the same as before, but by checking `undefined` of the first element instead of `.length` we can avoid the non-null assertion `first!`